src/main/java/com/solvd/laba/components/SuccessPopupComponent.java

```diff
@@ -0,0 +1,57 @@
+package com.solvd.laba.components;
+
+import java.time.Duration;
+
+
+import org.openqa.selenium.ElementClickInterceptedException;
+import org.openqa.selenium.JavascriptExecutor;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+import org.openqa.selenium.support.PageFactory;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.solvd.laba.pages.CartPage;
+
+public class SuccessPopupComponent {
+
+    private final WebDriver driver;
+
+    private static final Logger logger = LoggerFactory.getLogger(SuccessPopupComponent.class);
+
+    @FindBy(id = "cartModal")
+    private WebElement cartModalElement;
+
+    @FindBy(css = "a[href='/view_cart']")
+    private WebElement viewCartButton;
+
+    public SuccessPopupComponent(WebDriver driver) {
+        this.driver = driver;
+        PageFactory.initElements(driver, this);
+    }
+
+    public boolean isOpened() {
+        try {
+            WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(5));
+            wait.until(ExpectedConditions.visibilityOf(cartModalElement));
+            return cartModalElement.isDisplayed();
+        } catch (Exception e) {
```

Comment:
Catching TimeoutException in isOpened() is more precise than catching all Exception